engine/datastore/models.go

```diff
@@ -371,15 +370,12 @@ func (c *Client) Raw(query string) *gorm.DB {
 // checkResult will check for records or error
 func checkResult(result *gorm.DB) error {
 	if result.Error != nil {
-		if errors.Is(result.Error, gorm.ErrRecordNotFound) {
-			return ErrNoResults
-		}
```

Comment:
Yes, that's why I've changed the check in other places not to use the custom errors but gorms. With translate errors it's all handled for us and there is no need to return different error in this method. If the error will be ErrRecordNotFound it will be returned in return result.Error